src/main/java/com/ericsson/ei/mongo/MongoDBHandler.java

```diff
@@ -368,55 +374,75 @@ private boolean doDrop(String dataBaseName, String collectionName, MongoQuery qu
         }
 
         try {
-            List<String> collectionList = getCollectionList(databaseName);
-            if (!collectionList.contains(collectionName)) {
-                createCollection(databaseName, collectionName);
-            }
+            verifyExistanceOfCollection(databaseName, collectionName);
+
+            MongoDatabase db = mongoClient.getDatabase(databaseName);
+            MongoCollection<Document> collection = db.getCollection(collectionName);
+            return collection;
         } catch (MongoClientException e) {
             LOGGER.error("Failure when handling Mongo collection: {} ", e.getMessage(), e);
             return null;
```

Comment:
Can I be so mean to say that we should not catch the exception here and just catch it further out(if we dont want to log here, then we could rethrow.)? Then we get a bit of nice refactoring for this :)

Reply:
Mayor refactoring is needed for that and it is out of this scope.